# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../mdadm/64.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

when cross compiling against dietlibc:
sysfs.c: In function 'sysfs_read':
sysfs.c:81: warning: format '%d' expects type 'int', but argument 3 has type 'long unsigned int'

--- ./sysfs.c.orig	2008-03-11 00:17:59.000000000 +0000
+++ ./sysfs.c	2008-03-11 00:28:05.000000000 +0000
@@ -78,9 +78,9 @@
 		struct stat stb;
 		if (fstat(fd, &stb)) return NULL;
 		if (major(stb.st_rdev)==9)
-			sprintf(sra->name, "md%d", minor(stb.st_rdev));
+			sprintf(sra->name, "md%lu", minor(stb.st_rdev));
 		else
-			sprintf(sra->name, "md_d%d",
+			sprintf(sra->name, "md_d%lu",
 				minor(stb.st_rdev)>>MdpMinorShift);
 	} else {
 		if (devnum >= 0)
@@ -232,10 +232,10 @@
 	int n;
 	if (fstat(fd, &stb)) return 0;
 	if (major(stb.st_rdev) == 9)
-		sprintf(fname, "/sys/block/md%d/md/component_size",
+		sprintf(fname, "/sys/block/md%lu/md/component_size",
 			minor(stb.st_rdev));
 	else
-		sprintf(fname, "/sys/block/md_d%d/md/component_size",
+		sprintf(fname, "/sys/block/md_d%lu/md/component_size",
 			minor(stb.st_rdev)>>MdpMinorShift);
 	fd = open(fname, O_RDONLY);
 	if (fd < 0)
